lnwallet/size.go

```diff
@@ -186,68 +186,39 @@ const (
 	// weight limits.
 	MaxHTLCNumber = 966
 
-	// ToLocalScriptSize 83 bytes
+	// ToLocalScriptSize 79 bytes
 	//      - OP_IF: 1 byte
-	//              - OP_DATA: 1 byte (revocationkey length)
-	//              - revocationkey: 33 bytes
-	//              - OP_CHECKSIG: 1 byte
+	//          - OP_DATA: 1 byte
+	//          - revoke_key: 33 bytes
 	//      - OP_ELSE: 1 byte
-	//              - OP_DATA: 1 byte (localkey length)
-	//              - local_delay_key: 33 bytes
-	//              - OP_CHECKSIG_VERIFY: 1 byte
-	//              - OP_DATA: 1 byte (delay length)
-	//              - delay: 8 bytes
-	//              -OP_CHECKSEQUENCEVERIFY: 1 byte
+	//          - OP_DATA: 1 byte
+	//          - csv_delay: 4 bytes
+	//          - OP_CHECKSEQUENCEVERIFY: 1 byte
+	//          - OP_DROP: 1 byte
+	//          - OP_DATA: 1 byte
+	//          - delay_key: 33 bytes
 	//      - OP_ENDIF: 1 byte
-	ToLocalScriptSize = 1 + 1 + 33 + 1 + 1 + 1 + 33 + 1 + 1 + 8 + 1 + 1
-
-	// ToLocalTimeoutWitnessSize x bytes
-	//     - number_of_witness_elements: 1 byte
-	//     - local_delay_sig_length: 1 byte
-	//     - local_delay_sig: 73 bytes
-	//     - zero_length: 1 byte
-	//     - witness_script_length: 1 byte
-	//     - witness_script (to_local_script)
+	//      - OP_CHECKSIG: 1 byte
```

Comment:
The size of this one is correct according, just the script in the comment is wrong. The other ones were actually incorrect

Reply:
The spec says 8 bytes for the csv, but in practice this is more like one or two bytes as it's in relative block height. The spec doesn't account for the OP_DATA as we do here, where we allocate 1 + 4 bytes for the CSV. This is fine since it overestimates the total size anyway. I'd be fine with making it 8 as well just to match the spec.